### PR TITLE
Network utilities for all device types and to get interface name via hwaddrs

### PR DESCRIPTION
**Pull request by abdhaleegit**
This patch series adds
+ utilities added to identify different Network device interface 
type like virtual NIC, virther ethernet SEA, SRIOV logical interface etc and 
+  added utility to get interface names via unique hardware address 
+ With some pylint changes

### CODE REVIEW

File: avocado/utils/network/interfaces.py [hunk 1]
```diff
@@ -612,6 +612,52 @@ def is_bond(self):
...
 
+    def is_veth(self):
+        """Check if interface is a Virtual Ethernet.
+
+        This method checks if the interface is a Virtual Ethernet or not.
+
+        rtype: bool
+        """
+        if not os.path.isfile(f"/sys/class/net/{self.name}/device/devspec"):
+            raise ValueError("Please provide valid inerface name")
```
Comment (PraveenPenguin) [inline avocado/utils/network/interfaces.py:623]: handle this gracefully may  raise NWException kind of way

File: avocado/utils/network/interfaces.py [hunk 2]
```diff
@@ -612,6 +612,52 @@ def is_bond(self):
...
+        cmd = f"cat /sys/class/net/{self.name}/device/devspec"
+        output = run_command(cmd, self.host)
+        if "vnic" in output:
+            return True
+        return False
```
Comment (PraveenPenguin) [inline avocado/utils/network/interfaces.py:643]: can utility  is_vnic and is_veth combined name can be like 

```
is_virtualnet(self, mode=''):
     if mode='vnic':
        check_string= 'vnic'
     elif mode='veth':
          check_string= '"l-lan'
     else:
           raise NWException("Not supported Format")
  
   if not os.path.isfile(f"/sys/class/net/{self.name}/device/devspec"):
            raise NWException ("Please provide valid inerface name")
        cmd = f"cat /sys/class/net/{self.name}/device/devspec"
        output = run_command(cmd, self.host)
        if checkstring  in output:
            return True
        return False
  ```

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

File: avocado/utils/network/interfaces.py (hunk 2 above)
Comment (abdhaleegit) [reply avocado/utils/network/interfaces.py:643]: utility is only to know mode.. if it is already known.. than no need to check fro if it is vnic .. you got the point ?

Comment (abdhaleegit) [reply avocado/utils/network/interfaces.py:643]: with suggested way.. it is like you are passing vnic and returning vnic

Comment (PraveenPenguin) [reply avocado/utils/network/interfaces.py:643]: yes reasonable just looking to minimize the code uses may be common part can be push in another module

File: avocado/utils/network/interfaces.py (hunk 1 above)
Comment (abdhaleegit) [reply avocado/utils/network/interfaces.py]: Done

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**